### PR TITLE
Deprecate Customer::validateController

### PR DESCRIPTION
**Pull request by jf-viguier**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Deprecate Customer::validateController, it's not called anymore in the core. This method used to check password in 1.6
| Type?             | improvement
| Category?         | FO
| BC breaks?        | no
| Deprecations?     | yes
| Fixed ticket?     | https://github.com/PrestaShop/PrestaShop/issues/26293
| How to test?      | N/A
| Possible impacts? | None
| Sponsor company   | Creabilis.com

### CODE REVIEW

**Review by kpodemski — DISMISSED** [review]

File: classes/Customer.php [hunk 1]
```diff
@@ -1346,11 +1346,13 @@ public function getLastEmptyCart($withOrder = true)
...
 
     /**
-     * Validate controller.
+     * Validate controller and check password
      *
      * @param bool $htmlentities
      *
      * @return array
+     * 
+     * @deprecated The password check has been moved in controllers and this method is not called anywhere.
```
Comment (kpodemski) [inline classes/Customer.php]:
```suggestion
     * @deprecated Deprecated since PrestaShop 1.7.0
```

Comment (kpodemski) [reply classes/Customer.php]: I think that should do it

Comment (FabienPapet) [reply classes/Customer.php]: since PS 8.1 ? maybe

Comment (jf-viguier) [reply classes/Customer.php]: Hi guys @kpodemski @FabienPapet I added the deprecate, so it depends on the version you will accept the PR, so I can't guess it ? Please help.

Comment (FabienPapet) [reply classes/Customer.php]: I think as it is in develop, target is 8.1 as we can't deprecate features on released versions.

Comment (kpodemski) [reply classes/Customer.php]: Yes, obviously 8.1, sorry :D it is not used since 1.7, deprecated since 8.1 👍

Comment (jf-viguier) [reply classes/Customer.php]: @kpodemski @FabienPapet it's changed, thanks.

**Review by FabienPapet — APPROVED** [review]

**Review by matks — APPROVED** [review]

**Comment by matks** [comment]
Thank you @jf-viguier as this PR only modifies the code comments and no behavior, no QA is needed

**Comment by jf-viguier** [comment]
Thanks @matks

**Comment by thomasnares** [comment]
you can remove "needs documentation" label

**Pull request merged**